app/models/manageiq/providers/redhat/infra_manager/provision_workflow.rb

```diff
@@ -61,16 +61,6 @@ def allowed_customization_templates(options = {})
     end
   end
 
-  def allowed_hosts_obj(options = {})
```

Comment:
You are right. I re added it and changed it according to our needs, now it will choose only hosts from same datacenter.